app/forms/floating-ip-create.tsx

```diff
@@ -0,0 +1,157 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, you can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * Copyright Oxide Computer Company
+ */
+import { useMemo } from 'react'
+import { useWatch } from 'react-hook-form'
+import {
+  useNavigate,
+  type LoaderFunctionArgs,
+  type NavigateFunction,
+} from 'react-router-dom'
+
+import {
+  apiQueryClient,
+  useApiMutation,
+  useApiQuery,
+  useApiQueryClient,
+  type FloatingIp,
+  type FloatingIpCreate,
+  type SiloIpPool,
+} from '@oxide/api'
+import { validateIp } from '@oxide/util'
+
+import {
+  DescriptionField,
+  ListboxField,
+  NameField,
+  SideModalForm,
+  TextField,
+} from 'app/components/form'
+import { getProjectSelector, useForm, useProjectSelector, useToast } from 'app/hooks'
+import { pb } from 'app/util/path-builder'
+
+CreateFloatingIpSideModalForm.loader = async ({ params }: LoaderFunctionArgs) => {
+  await apiQueryClient.prefetchQuery('projectIpPoolList', {
+    query: { ...getProjectSelector(params), limit: 1000 },
+  })
+  return null
+}
+
+type CreateSideModalFormProps = {
+  /**
+   * If defined, this overrides the usual mutation. Caller is responsible for
+   * doing a dismiss behavior in onSubmit as well, because we are not calling
+   * the RQ `onSuccess` defined for the mutation.
+   */
+  onSubmit?: (floatingIpCreate: FloatingIpCreate) => void
+  /**
+   * Passing navigate is a bit of a hack to be able to do a nav from the routes
+   * file. The callers that don't need the arg can ignore it.
+   */
+  onDismiss?: (navigate: NavigateFunction) => void
+  onSuccess?: (floatingIp: FloatingIp) => void
+}
```

Comment:
I'd get rid of these props. Some other forms have them because they get loaded in multiple contexts, like how you can create a disk (sort of virtually) from the instance create form in addition to the disks list. Forms that get called in a single context don't need these props because they can hard code their behavior. Some examples:

https://github.com/oxidecomputer/console/blob/d4ec192725e7410c2f5b4d653fc223eb04f0569b/app/forms/ip-pool-create.tsx#L22

https://github.com/oxidecomputer/console/blob/d4ec192725e7410c2f5b4d653fc223eb04f0569b/app/forms/silo-create.tsx#L51

https://github.com/oxidecomputer/console/blob/d4ec192725e7410c2f5b4d653fc223eb04f0569b/app/forms/ip-pool-create.tsx#L22